### PR TITLE
PipelineRunSummary should include a change set

### PR DESCRIPTION
**Pull request by stuartrowe**
# Description

See [JENKINS-55497](https://issues.jenkins-ci.org/browse/JENKINS-55497).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [X] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by stuartrowe** [comment]
@sophistifunk @halkeye please review when you have a chance.

**Comment by halkeye** [comment]
oh nice! @NicuPascu and I have been looking into something similar, wonder if it fixes things.

I'll do a proper review && merge on monday (unless josh gets to it first) but can you add a simple test? https://github.com/jenkinsci/blueocean-plugin/pull/1917/files#diff-9fcb2682093f52b0f9d0f372f330bf33R1 should be a good starting point.

**Comment by stuartrowe** [comment]
@halkeye thanks for providing a test to reference, that really helps.

However, I'm not sure how easy it will be to demonstrate the issue in a test like this. The REST endpoint `/organizations/jenkins/pipelines/<project>/runs/<id>/` already correctly returns a change set. The issue is a caching a problem on the javascript/react end which is outside my comfort zone. There are details in a comment on the issue:
https://issues.jenkins-ci.org/browse/JENKINS-55497?focusedCommentId=359998&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-359998

So, do you have any suggestions for retrieving a `PipelineRunSummary`? Perhaps I can manufacture a stapler request for this?

File: blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineImplTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        String jenkinsFile = Resources.toString(Resources.getResource(getClass(), "singleScm.jenkinsfile"), Charsets.UTF_8).replaceAll("%REPO%", sampleRepo.toString());
+
+        WorkflowJob p = j.createProject(WorkflowJob.class, "project");
+        p.setDefinition(new CpsFlowDefinition(jenkinsFile, true));
+        p.save();
+
+        sampleRepo.init();
+
+        Run r = p.scheduleBuild2(0).waitForStart();
+        j.waitForCompletion(r);
```
Comment (halkeye) [inline blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineImplTest.java]: I don't know how to make a github suggestion, but I'd change it from:
```
        Run r = p.scheduleBuild2(0).waitForStart();
        j.waitForCompletion(r);
```

to

```
        Run r = j.assertBuildStatus(Result.SUCCESS, p.scheduleBuild2(0));
```

Cause on the other PR, I didn't notice the typo in the script file, which was causing the test to fail in ways i couldn't easily see.

Comment (stuartrowe) [reply blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineImplTest.java]: Updated, thanks for the suggestion.

**Comment by halkeye** [comment]
@stuartrowe looks like your have an import error. I'm mobile so can't help more right now

We are going to try and finish merging things and do a release tomorrow. If tests are passing we will include this, otherwise it'll be in the next one.

**Comment by stuartrowe** [comment]
The import error should be sorted out now. I've been having a lot of trouble building lately. I'll ping you direct on gitter tomorrow with a few related questions if that's alright.

**Comment by halkeye** [comment]
sure, wed are pretty good for me.

I think the tests are failing because first build never has a changelog. It needs an initial state (first build) to have changes from (second build)

**Comment by stuartrowe** [comment]
Yes, I'll look into fixing that later today.

**Comment by stuartrowe** [comment]
The ATH failed tests don't seem related as the most recent commit just updated a _different_ test.

**Comment by halkeye** [comment]
yea, we are having issue with ath right now. 

Thanks for your contribution, we'll get it merged asap

**Comment by halkeye** [comment]
Awesome. This totally works on my machine. I setup a test on ci.blueocean.io to confirm the old behavior.

Merging!

**Pull request merged**